allennlp/modules/seq2vec_encoders/cnn_encoder.py

```diff
@@ -0,0 +1,119 @@
+from typing import Optional, Tuple
+
+from overrides import overrides
+import torch
+from torch.nn import Conv1d, Linear
+
+from allennlp.common import Params
+from allennlp.experiments import Registry
+from allennlp.modules import Seq2VecEncoder
+
+@Registry.register_seq2vec_encoder("cnn")
+class CnnEncoder(Seq2VecEncoder):
+    """
+    CnnEncoder is a combination of multiple convolution layers and max pooling layers.  As a
+    :class:`Seq2VecEncoder`, the input to this module is of shape ``(batch_size, num_tokens,
+    input_dim)``, and the output is of shape ``(batch_size, output_dim)``.
+
+    The CNN has one convolution layer for each ngram filter size. Each convolution operation gives
+    out a vector of size num_filters. The number of times a convolution layer will be used
+    is ``num_tokens - ngram_size + 1``. The corresponding maxpooling layer aggregates all these
+    outputs from the convolution layer and outputs the max.
+
+    This operation is repeated for every ngram size passed, and consequently the dimensionality of
+    the output after maxpooling is ``len(ngram_filter_sizes) * num_filters``.  This then gets
+    (optionally) projected down to a lower dimensional output, specified by ``output_dim``.
+
+    We then use a fully connected layer to project in back to the desired output_dim.  For more
+    details, refer to "A Sensitivity Analysis of (and Practitioners’ Guide to) Convolutional Neural
+    Networks for Sentence Classification", Zhang and Wallace 2016, particularly Figure 1.
+
+    Parameters
+    ----------
+    embedding_dim : ``int``
```

Comment:
Are we doing ``int`` or just "int"? I've just been using "int", but you have both here.